### PR TITLE
New feature: LoggingCallback

### PR DESCRIPTION
**Pull request by bstriner**
Hi everybody!

This is a callback that I've been using and thought other people might find handy. This PR adds a LoggingCallback that prints an end-of-epoch summary. The print function is a parameter, so it can print the summary to Python logging or a file. Each epoch it calls the given print function with a formatted summary. The format of the summary is controlled by parameters to LoggingCallback.

 ```python
# Write to Python logging
import logging
model.fit(x, y, callbacks = [LoggingCallback(logging.info)]

# Write to a file
with open("log.txt", 'w') as f:
  def print_fcn(s):
    f.write(s)
    f.write("\n")
  model.fit(x, y, callbacks = [LoggingCallback(print_fcn)]
```

The summary itself looks like the below by default but the format strings are all configurable.

> Epoch: 3 - loss: 3.4123 - val_loss: 5.4321

Cheers,
Ben

### CODE REVIEW

**Comment by bstriner** [comment]
Also, added a unit test that checks that the callback fires and calls the print function. Doesn't actually test that the format looks good.

**Comment by fchollet** [comment]
I think this should go into the contrib repo for the time being.
